File: test/Test_BisonMeat.cpp
#include "../include/BisonMeat.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(BisonMeatTest, AllFunctionalityofBisonMeat){
    BisonMeat testBM(75, "Macho");
    ASSERT_EQ("Macho", testBM.getProducer());
    ASSERT_EQ(BISON_MEAT, testBM.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testBM.getProductType());
    ASSERT_EQ(75, testBM.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/meat/TestBisonMeat.java
package engisubs.gameobject.product.farmproduct.meat;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestBisonMeat{
    private BisonMeat test;
    @Before
    public void initObject(){
        test = new BisonMeat(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", test.getProducer());
        Assert.assertEquals(GameObjectType.BISON_MEAT, test.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(25, test.getPrice());
    }

}